File: for-local-test/python/326.power-of-three.py
class Solution:
    def isPowerOfThree(self, n: int) -> bool:
        maxPowerOfThree = 3 ** 19
        return n > 0 and maxPowerOfThree % n == 0

if __name__ == "__main__":
    solution = Solution()
    print(solution.isPowerOfThree(27))  # true
    print(solution.isPowerOfThree(0))   # false
    print(solution.isPowerOfThree(9))   # true
    print(solution.isPowerOfThree(45))  # false

File: for-local-test/326.power-of-three.java
class Solution {
    public boolean isPowerOfThree(int n) {
        int maxPowerOfThree = (int) Math.pow(3, 19);
        return n > 0 && maxPowerOfThree % n == 0;
    }
    public static void main(String[] args) {
        Solution solution = new Solution();
        System.out.println(solution.isPowerOfThree(27)); // true
        System.out.println(solution.isPowerOfThree(0));  // false
        System.out.println(solution.isPowerOfThree(9));  // true
        System.out.println(solution.isPowerOfThree(45)); // false
    }
}


